### PR TITLE
MINOR: Make JUnit 5 the default for new projects

### PR DESCRIPTION
**Pull request by ijuma**
Instead of listing the projects that should use JUnit 5, list the
projects that are still using JUnit 4.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by ijuma** [comment]
Merge https://github.com/apache/kafka/pull/9856 first.

File: build.gradle [hunk 1]
```diff
@@ -241,7 +241,10 @@ subprojects {
     }
   }
 
-  def shouldUseJUnit5 = ["clients", "examples", "log4j-appender", "raft", "test-utils", "tools"].contains(it.project.name)
+  // Remove the relevant project name once it's converted to JUnit 5
+  def shouldUseJUnit5 = !(["api", "basic-auth-extension", "connect", "core", "file", "generator",
+    "json", "mirror", "mirorr-client", "runtime", "transform", "examples", "streams-scala",
```
Comment (chia7712) [inline build.gradle:246]: ```examples``` is already in junit 5 (bed4c6a33b16b96c1a20f47d818f7a44eea33fa9)

**Review by chia7712 — APPROVED** [review]
+ 1 if the following suggestion is addressed.

File: build.gradle (hunk 1 above)
Comment (chia7712) [reply build.gradle:246]: oh, there is another module called ```examples```

Comment (ijuma) [reply build.gradle:246]: It doesn't have tests though, so I can probably remove it. Let me check.

Comment (ijuma) [reply build.gradle:246]: Fixed.

**Pull request merged**